GVFS/FastFetch/CheckoutPrefetcher.cs

```diff
@@ -72,87 +72,104 @@ public override void Prefetch(string branchOrCommit, bool isBranch)
                 commitToFetch = branchOrCommit;
             }
 
-            this.DownloadMissingCommit(commitToFetch, this.GitObjects);
+            if (!IndexLock.TryGetLock(this.Enlistment.EnlistmentRoot, this.Tracer, out var indexLock))

```

Comment:
Not to any other reviewers: this diff looks a lot better if you change the diff options to "Hide Whitespace".